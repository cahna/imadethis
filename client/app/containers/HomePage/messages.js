/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.containers.HomePage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Welcome to Enterprise-Standard Chat!',
  },
  loginHeader: {
    id: `${scope}.login.message`,
    defaultMessage: 'Login to begin',
  },
  loginUsernameLabel: {
    id: `${scope}.loginUsernameLabel.message`,
    defaultMessage: 'Enter your username',
  },
  submitLoginForm: {
    id: `${scope}.submitLoginForm.message`,
    defaultMessage: 'Login',
  },
});
