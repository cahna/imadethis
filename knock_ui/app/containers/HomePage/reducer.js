/*
 * HomeReducer
 *
 * The reducer takes care of our data. Using actions, we can
 * update our application state. To add a new action,
 * add it to the switch statement in the reducer function
 *
 */

import produce from 'immer';
import { CHANGE_USERNAME, CHANGE_THREAD } from './constants';

// The initial state of the App
export const initialState = {
  username: '',
  threadId: null,
};

/* eslint-disable default-case, no-param-reassign */
const homeReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case CHANGE_USERNAME:
        // Remove non-alphanumeric characters
        draft.username = action.username.replace(/[^0-9a-z]/gi, '');
        break;
      case CHANGE_THREAD:
        draft.threadId = action.threadId;
        break;
    }
  });

export default homeReducer;
