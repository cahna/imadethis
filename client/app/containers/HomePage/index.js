/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
// import { push } from 'connected-react-router';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

// import { useInjectReducer } from 'utils/injectReducer';
// import { useInjectSaga } from 'utils/injectSaga';
import H2 from 'components/H2';
import {
  makeSelectCurrentUser,
  makeSelectLoading,
  makeSelectError,
} from 'containers/App/selectors';

import Section from './Section';
import messages from './messages';
// import reducer from './reducer';
// import saga from './saga';

// const key = 'home';

export function HomePage({
  currentUser,
  // loading,
  // error,
  // dispatch,
}) {
  // useInjectReducer({ key, reducer });
  // useInjectSaga({ key, saga });

  return (
    <article>
      <Helmet>
        <title>Enterprise-Standard Chat</title>
        <meta name="description" content="Chat for tryhards" />
      </Helmet>
      <div>
        <Section>
          <h1>
            <FormattedMessage {...messages.header} />
          </h1>
        </Section>
        <Section>
          <H2>Hello, {currentUser.username}!</H2>
        </Section>
      </div>
    </article>
  );
}

HomePage.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
  currentUser: PropTypes.shape({
    username: PropTypes.string,
    uniqueId: PropTypes.string,
  }).isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  currentUser: makeSelectCurrentUser(),
  loading: makeSelectLoading(),
  error: makeSelectError(),
});

export function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  memo,
)(HomePage);
