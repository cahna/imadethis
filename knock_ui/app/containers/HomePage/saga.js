/**
 * Gets the threads of the user from API
 */

import { call, put, select, takeLatest } from 'redux-saga/effects';
import { LOAD_USER_THREADS } from 'containers/App/constants';
import {
  userThreadsLoaded,
  userThreadLoadingError,
} from 'containers/App/actions';

import request from 'utils/request';
import { makeSelectUsername } from 'containers/HomePage/selectors';

/**
 * Github repos request/response handler
 */
export function* getUserThreads() {
  // Select username from store
  const username = yield select(makeSelectUsername());
  const requestURL = `https://localhost:3000/users/${username}`;

  try {
    // Call our request helper (see 'utils/request')
    const userThreads = yield call(request, requestURL);
    yield put(userThreadsLoaded(userThreads, username));
  } catch (err) {
    yield put(userThreadLoadingError(err));
  }
}

/**
 * Root saga manages watcher lifecycle
 */
export default function* userThreadsData() {
  // Watches for LOAD_THREADS actions and calls getThreads when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(LOAD_USER_THREADS, getUserThreads);
}
