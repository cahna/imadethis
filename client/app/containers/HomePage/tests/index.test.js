import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';

import history from 'utils/history';

import HomePage from '../index';
import configureStore from '../../../configureStore';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, history);
  });

  it('should render and match the snapshot', () => {
    const doLogoutUser = jest.fn();

    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage
            loading={false}
            error={false}
            currentUser={{
              username: 'TestUser',
              uniqueId: 'abc-123',
            }}
            doLogoutUser={doLogoutUser}
          />
        </IntlProvider>
      </Provider>,
    );

    expect(doLogoutUser).not.toHaveBeenCalled();
    expect(firstChild).toMatchSnapshot();
  });
});
