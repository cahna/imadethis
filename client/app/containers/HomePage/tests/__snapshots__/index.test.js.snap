// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<article>
  <div>
    <section
      class="Section-ojdkzv-0 CenteredSection-sc-1e2by6n-0 jhRZyk"
    >
      <h1>
        <span>
          What have you made?
        </span>
      </h1>
    </section>
    <section
      class="Section-ojdkzv-0 CenteredSection-sc-1e2by6n-0 jhRZyk"
    >
      <h2
        class="H2-egphza-0 HsCXl"
      >
        Hello, 
        !
      </h2>
    </section>
    <section
      class="Section-ojdkzv-0 CenteredSection-sc-1e2by6n-0 jhRZyk"
    >
      <div
        class="Wrapper-sc-5dsnpv-0 jpYyqh"
      >
        <a
          class="A-t53uvo-0 imPlZY"
        >
          <span>
            Logout
          </span>
        </a>
      </div>
    </section>
  </div>
</article>
`;
